clang/lib/Driver/ToolChains/HLSL.cpp

```diff
@@ -295,6 +295,32 @@ void tools::hlsl::MetalConverter::ConstructJob(
                                          Exec, CmdArgs, Inputs, Input));
 }
 
+void tools::LLVMObjcopy::ConstructJob(Compilation &C, const JobAction &JA,
+                                      const InputInfo &Output,
+                                      const InputInfoList &Inputs,
+                                      const ArgList &Args,
+                                      const char *LinkingOutput) const {
+
+  std::string ObjcopyPath = getToolChain().GetProgramPath("llvm-objcopy");
```

Comment:
don't work much in the driver, should we be using a std:string here, I've noticed this file uses StringRef in other places.

Reply:
`GetProgramPath` returns a constructed string as a `std::string`, so you need to take ownership of the allocation. Using a StringRef to a returned temporary would be bad.